packages/api/src/models/animalBatchModel.js

```diff
@@ -51,9 +51,21 @@ class AnimalBatchModel extends baseModel {
         id: { type: 'integer' },
         name: { type: 'string' },
         notes: { type: ['string', 'null'] },
+        removed: { type: 'boolean' },
+        animal_removal_reason_id: { type: ['integer', 'null'] },
+        removal_explanation: { type: ['string', 'null'] },
         ...this.baseProperties,
       },
       additionalProperties: false,
+      if: {
+        properties: {
+          removed: { const: true },
+        },
+        required: ['removed'],
+      },
+      then: {
+        required: ['animal_removal_reason_id'],
+      },
```

Comment:
This if-then on the schema is only triggering on POST (which was the only controller that existed at the time I wrote this) and not on PATCH. Therefore it's not a useful approach (and not my favourite syntax either) soI'll remove it in the PATCH PR, in favour of either some other custom model function or just sticking with the database validation.